library/src/main/java/com/firebase/ui/auth/google/GoogleAuthProvider.java

```diff
@@ -93,7 +82,7 @@ public void onResult(Status status) {
     public void login() {
         if (mGoogleApiClient.isConnected()) {
             Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
-            mActivity.startActivityForResult(signInIntent, GoogleActions.SIGN_IN);
+            ((Activity)getContext()).startActivityForResult(signInIntent, GoogleActions.SIGN_IN);
```

Comment:
I pulled the mActivity member into the base class and gave that a
getContext() accessor method. That might have been in a previous refactor
already, with this change just getting rid of the mActivity member.

I am a bit torn on getContext(). Since we seem to most often want to use it
as an Activity, we might just keep it as an Activity and prevent the need
to downcast.

On Wed, Dec 30, 2015 at 10:51 AM Abe Haskins <notifications@github.com>
wrote:

> In
> library/src/main/java/com/firebase/ui/auth/google/GoogleAuthProvider.java
> <https://github.com/firebase/FirebaseUI-Android/pull/47#discussion_r48623211>
> :
>
> > @@ -93,7 +82,7 @@ public void onResult(Status status) {
> >      public void login() {
> >          if (mGoogleApiClient.isConnected()) {
> >              Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
> > -            mActivity.startActivityForResult(signInIntent, GoogleActions.SIGN_IN);
> > +            ((Activity)getContext()).startActivityForResult(signInIntent, GoogleActions.SIGN_IN);
>
> Curious about the reasoning for the change here, I'm assuming this
> functioned before, so why the change?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/firebase/FirebaseUI-Android/pull/47/files#r48623211>.
>